domain/sysvar_cache.go

```diff
@@ -162,6 +164,31 @@ func checkEnableServerGlobalVar(name, sVal string) {
 		err = stmtsummary.StmtSummaryByDigestMap.SetMaxSQLLength(sVal, false)
 	case variable.TiDBCapturePlanBaseline:
 		variable.CapturePlanBaseline.Set(sVal, false)
+	case variable.TiDBEnableTopSQL:
+		config.UpdateGlobal(func(conf *config.Config) {
+			conf.TopSQL.Enable = variable.TiDBOptOn(sVal)
+		})
+	case variable.TiDBTopSQLAgentAddress:
+		config.UpdateGlobal(func(conf *config.Config) {
+			// todo: add validate
+			conf.TopSQL.AgentAddress = sVal
+		})
+	case variable.TiDBTopSQLPrecisionSeconds:
+		var val int
+		val, err = strconv.Atoi(sVal)
+		if err == nil {
+			config.UpdateGlobal(func(conf *config.Config) {
+				conf.TopSQL.PrecisionSeconds = val
+			})
+		}
+	case variable.TiDBTopSQLMaxStatementCount:
+		var val int
+		val, err = strconv.Atoi(sVal)
+		if err == nil {
+			config.UpdateGlobal(func(conf *config.Config) {
+				conf.TopSQL.MaxStatementCount = val
+			})
+		}
```

Comment:
This is the same comment as with executor/set.go - I would like to eventually remove this code and rely on getters and setters.

The `GetSession` function is "special" in that it is also called when a new session is Inited if that helps (unless `skipInit` is set).

Reply:
Great, already changed, PTAL.